PythonRegression/tests/features/steps/blowball_tests_steps.py

```diff
@@ -0,0 +1,89 @@
+from aloe import world, step 
+from util.test_logic import api_test_logic
+from iota import *
+
+import logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+tests = api_test_logic
+
+config = {}
+
+
+@step(r'find transaction is called with the address "([^"]*)" to find milestones')
+def find_transactions_from_coo(step,address):
+    logger.info('Finding milestones')
+    machine = 'machine3'
+    node = 'nodeA'       
+    config['node'] = node
+    
+    api = tests.prepare_api_call(node, machine)
+    config['find_transactions_response'] = api.find_transactions(addresses = [address]) 
+
+
+@step(r'the response should contain at least (\d+) transactions')
+def check_for_milestone_number(step, numTransactions):
+    response_length = len(config['find_transactions_response']['hashes'])
+    assert response_length >= 5, response_length
+    logger.info(config['find_transactions_response'])
+
+
+
+@step(r'Then the returned GTTA transactions will be compared with the milestones')
+def compare_gtta_with_milestones(step):
+    logger.info("compare")
+    gtta_responses = tests.fetch_response('getTransactionsToApprove')
+    milestones = list(config['find_transactions_response']['hashes'])
+    
+    config['max'] = len(gtta_responses[config['node']])
+    
+    transactions = []
+    transactions_count = []
+    config['milestone_count'] = 0
+    
+    for i in gtta_responses:
+        if type(gtta_responses[i]) is list:
+            for x in range(len(gtta_responses[i])):
+                if gtta_responses[i][x]['branchTransaction'] not in milestones:
+                    if gtta_responses[i][x]['branchTransaction'] not in transactions:
+                        transactions.append(gtta_responses[i][x]['branchTransaction']) 
+                        transactions_count.append(1)
+                        logger.info('added transaction')
+                    else: 
+                        transactions_count[transactions.index(gtta_responses[i][x]['branchTransaction'])] += 1                                    
+                else: 
+                    config['milestone_count'] += 1
+                    logger.info('milestone')
+                             
+                             
+                if gtta_responses[i][x]['trunkTransaction'] not in milestones:
+                    if gtta_responses[i][x]['trunkTransaction'] not in transactions:
+                        transactions.append(gtta_responses[i][x]['trunkTransaction'])
+                        transactions_count.append(1)
+                        logger.info('added transaction') 
+                    else:
+                        transactions_count[transactions.index(gtta_responses[i][x]['trunkTransaction'])] += 1                
+                else: 
+                    config['milestone_count'] += 1
+                    logger.info('milestone')
+    
```

Comment:
all those nested `for` and `ifs` are confusing. I didn't even bother to try to understand the code. Can you break up to smaller functions and add comments?